corehq/apps/hqwebapp/views.py

```diff
@@ -541,7 +542,10 @@ def ping_response(request):
         app = get_app_cached(domain, current_build_id)
         app_id = app['copy_of'] if app['copy_of'] else app['_id']
         latest_build_id = get_latest_released_build_id(domain, app_id)
-        if latest_build_id:
+        # Do not show popup to users who have use_latest_build_cloudcare ff enabled
+        latest_build_ff_enabled = (CLOUDCARE_LATEST_BUILD.enabled(domain)
```

Comment:
It might be nice to move this check above the couch request, so we can skip the query entirely if we don't need it.

Reply:
This would be a natural place to use `CLOUDCARE_LATEST_BUILD.enabled_for_request`

Reply:
> This would be a natural place to use CLOUDCARE_LATEST_BUILD.enabled_for_request

@orangejenny  I started with that only but when I was testing on my local, I figured out the request object does not have domain info available because of which `CLOUDCARE_LATEST_BUILD.enabled_for_request` was returning `False`. So I would have to manually check it here.

Reply:
Interesting - I wonder if it's because `ping_response` doesn't using `login_and_domain_required`?

Reply:
Not exactly sure if `login_and_domain_required` sets domain on request. 
My guess is that the domain-specific URLs https://github.com/dimagi/commcare-hq/blob/master/urls.py#L107 have `request.domain` but I can be wrong it that.

Reply:
ah, that makes more sense